Add --refresh flag to models command

ahosker opened the pull request:
## Summary
- Add `--refresh` flag to the `opencode models` command to refresh the models cache
- When used, calls `ModelsDev.refresh()` to force fetch fresh data from models.dev (an existing function in the codebase)
- Shows a success message after cache refresh
- Allows users to refresh their cached models when needed manually
- Shows models after refreshing

## Usage
```bash
opencode models --refresh
```
or
```bash
opencode models opencode --refresh
```

This is useful when users want to force refresh their models cache to get the latest models from models.dev without waiting for the automatic hourly refresh. Such as an hour ago, when a new Opus was released.

Currently, the advice is to run `rm ~/.cache/opencode/models.json`, which is equally as effective but less cool.

ahosker committed:
Add --clear flag to models command to refresh cache

- Add --clear option to models CLI command
- When used, calls ModelsDev.refresh() to update cache
- Shows success message after cache refresh
- Allows users to force refresh models from models.dev

rekram1-node commented:
prolly should call it --refresh

ahosker committed:
Rename --clear flag to --refresh for better clarity

- Change CLI flag from --clear to --refresh
- Update description to better reflect functionality
- Update success message to match new flag name

ahosker committed:
Merge branch 'dev' into feature/models-clear-cache

rekram1-node commented:
I'm also thinking since it is a flag it should still output the model list, otherwise a subcommand would make more sense prolly

/oc update this so that if it refreshes the model list it still outputs the models afterwards

ahosker committed:
Make --refresh flag show models after refreshing

- Remove early return when --refresh flag is used
- Now refreshes cache and then displays models list
- Follows standard CLI flag behavior where flags modify main action

ahosker commented:
True - taking that one line away is an improvement for sure, can confirm if the model exists that way also
```
opencode models github-copilot --refresh
```

rekram1-node commented:
Lol I tried to use the github action but u were faster

rekram1-node commented:
thanks!
